wd_helper.au3

```diff
@@ -1038,27 +1067,55 @@ Func _WD_ElementSelectAction($sSession, $sSelectElement, $sCommand)
 						$vResult = $aAllOptions
 					EndIf
 
+				Case 'selectAll'
+					$sScript = "var options = arguments[0].options; for ( i=0; i<options.length; i++) {options[i].selected = 'true';}; return true;"
+					$vResult = _WD_ExecuteScript($sSession, $sScript, __WD_JsonElement($sSelectElement), Default, $_WD_JSON_Value)
+					$iErr = @error
+
 				Case 'selectedIndex'
 					$sScript = "return arguments[0].selectedIndex"
 					$vResult = _WD_ExecuteScript($sSession, $sScript, __WD_JsonElement($sSelectElement), Default, $_WD_JSON_Value)
 					$iErr = @error
 
+				Case 'selectedLabels'
+					$sScript = "var result =''; var options = arguments[0].selectedOptions; for (let i = 0; i < options.length; i++) {result += options[i].label + '\n'} return result;"
+					$vResult = _WD_ExecuteScript($sSession, $sScript, __WD_JsonElement($sSelectElement), Default, $_WD_JSON_Value)
+					$iErr = @error
+
+					If $iErr = $_WD_ERROR_Success Then
+						Local $aSelectedLabels[0]
+						_ArrayAdd($aSelectedLabels, StringStripWS($vResult, $STR_STRIPTRAILING), 0, @LF, "", $ARRAYFILL_FORCE_DEFAULT)
+						$vResult = $aSelectedLabels
+					EndIf
+
+				Case 'selectedOptions'
+					$sScript = "var result =''; var options = arguments[0].selectedOptions; for (let i = 0; i < options.length; i++) {result += options[i].value + '|' + options[i].label + '|' + options[i].index + '|' + options[i].selected + '\n'} return result;"
+					$vResult = _WD_ExecuteScript($sSession, $sScript, __WD_JsonElement($sSelectElement), Default, $_WD_JSON_Value)
+					$iErr = @error
+
+					If $iErr = $_WD_ERROR_Success Then
+						Local $aSelectedOptions[0][4]
+						_ArrayAdd($aSelectedOptions, StringStripWS($vResult, $STR_STRIPTRAILING), 0, Default, @LF, $ARRAYFILL_FORCE_SINGLEITEM)
+						$vResult = $aSelectedOptions
+					EndIf
+
 				Case 'value'
 					$sScript = "return arguments[0].value"
 					$vResult = _WD_ExecuteScript($sSession, $sScript, __WD_JsonElement($sSelectElement), Default, $_WD_JSON_Value)
 					$iErr = @error
 
 				Case Else
-					Return SetError(__WD_Error($sFuncName, $_WD_ERROR_InvalidDataType, "(options|selectedIndex|value) $sCommand=>" & $sCommand), 0, "")
+					Return SetError(__WD_Error($sFuncName, $_WD_ERROR_InvalidDataType, "(deselectAll|multiSelect|options|selectAll|selectedIndex|selectedLabels|selectedOptions|value) $sCommand=>" & $sCommand), 0, "")
 
 			EndSwitch
 		Else
 			$iErr = $_WD_ERROR_InvalidArgue
+			$iExt = 49 ; $iExt from 41 to 49 are related to _WD_ElementSelectAction()
 		EndIf
 	EndIf
 
 	If $_WD_DEBUG = $_WD_DEBUG_Info Then
-		__WD_ConsoleWrite($sFuncName & ': ' & ((IsArray($vResult)) ? "(array)" : $vResult))
+		__WD_ConsoleWrite($sFuncName & ': $sCommand = ' & $sCommand & ' : Error = ' & $iErr & ' : Extended = ' & $iExt & ' : Result = ' & ((IsArray($vResult)) ? "(array)" : $vResult))
```

Comment:
You've added a lot of details here. Is all of this needed given upcoming changes to __WD_Error?

Reply:
Take a look on: [_WD_ElementSelectAction() $sMessage refactoring](https://github.com/Danp2/au3WebDriver/pull/212/commits/122fbcd241e097cbdc131dda8c8cf530dbbf1338)

What you think about the concept to move all this
```autoit
If $_WD_DEBUG = $_WD_DEBUG_Info Then
```

to internall processing into: 
```AutoIt
__WD_Error($sFuncName, $iErr, $sMessage)
```

If you agree then I open separate ISSUE to review all other codes to suplement this kind of processing.

Reply:
Yes, that's a good idea.

Reply:
Whats about this PR ?
Just wait to review ?